### PR TITLE
Use actual batch size for resize estimates

### LINKED ISSUE #2450: monero-blockchain-import v0.11.0.0 ERROR "MDB_MAP_FULL: Environment mapsize limit reached"
I tried to import the blockchain into v0.11.0.0, after exporting it from from v0.10.3.1. The host is a GNU/Linux amd64, 12 Gb RAM, over 120 Gb of free disk on SATA III HD.

During the process, the host becomes almost unresponsive, the 4 cores went overwhelmed by more of 10 pending processes. Some suggestion on how to help in debugging? You can see the error in the log attached.

[monero-blockchain-import.log.txt](https://github.com/monero-project/monero/files/1305848/monero-blockchain-import.log.txt)

### PR DESCRIPTION
**Pull request by hyc**
Fix #2450

### CODE REVIEW

File: src/blockchain_utilities/blockchain_import.cpp [hunk 1]
```diff
@@ -294,6 +292,27 @@ int import_from_file(cryptonote::core& core, const std::string& import_file_path
   // at start_height.
   while (! quit)
   {
+    if (use_batch && !(h % db_batch_size))
+    {
+      uint64_t bytes;
+      uint64_t h2 = 0;
+      bool q2 = false;
+      if (h)
```
Comment (moneromooo-monero) [inline src/blockchain_utilities/blockchain_import.cpp]: I'm not sure that works when restarting on a partially imported file: h % x may be non zero, and so batch_stop without a batch_start ?

File: src/cryptonote_core/blockchain.cpp [hunk 2]
```diff
@@ -3666,6 +3666,19 @@ bool Blockchain::prepare_handle_incoming_blocks(const std::list<block_complete_e
   MTRACE("Blockchain::" << __func__);
   TIME_MEASURE_START(prepare);
   bool stop_batch;
+  uint64_t bytes = 0;
+
+  if(blocks_entry.size() == 0)
+    return false;
```
Comment (moneromooo-monero) [inline src/cryptonote_core/blockchain.cpp]: That seems to return before taking the tx_lock, which cleanup_handle_x_x will unlock (regardless of what prepare_x_x_x returns).

**Comment by hyc** [comment]
OK, reverting to original control flow...

**Comment by moneromooo-monero** [comment]
Can you squash though please ?

**Comment by hyc** [comment]
Squashed. Should we fix this in release branch too?

**Comment by moneromooo-monero** [comment]
If pony releases a new version, sure.

**Pull request merged**